### DISCUSSION TITLE
Nano CouchDB Library: Error happened in your connection

### DISCUSSION THREAD

**Discussion by JeffreyQin** [apache/couchdb#4711] [Q&A] [answered]
I'm trying to incorporate CouchDB into a Node.js backend using the Nano npm library. According to the docs, the connection should be established using

```
const nano = require('nano')('http://localhost:5984');
```
When I run the app, however, it returns to me the error 
```
Error: error happened in your connection. Reason: connect ECONNREFUSED ::1:5984
```
The docs also indicated that the url may need to contain credentials in the form 'http:admin:password@localhost:5984', but what exactly are the 'admin' and 'password' referring to? Are they the login credentials for my fauxton portal?

**Comment by nickva** [comment]
@JeffreyQin yeah it can be admin user. Here is how to add a new user: https://docs.couchdb.org/en/stable/intro/security.html#creating-a-new-admin-user

At some point in the past (in the 1.x, 2.x era), there was no admin user required by default. That's why there are guides and examples floating around which may not mention it. That all changed in 3.0 if you're interested to learn more: https://blog.couchdb.org/2020/02/26/the-road-to-couchdb-3-0-security/

**Comment by rnewson** [comment, marked as answer]
`ECONNREFUSED ::1:5984` is a networking issue, you didn't get as far as making a request, with or without credentials.

sounds like couchdb is simply not running on the server you are using nano on, or it is not bound to the loopback interface, or it is bound only to ipv4 and you're contacting over ipv6 (::1 is ipv6 for localhost)

**Comment by JeffreyQin** [comment]
Thank you guys for the answers. I was able to resolve the issue and successfully connect to CouchDB by performing the following changes
1. changing the url 'http://localhost:5984' to 'http://127.0.0.1:5984' to resolve version issues
2. provide my admin and password so the url looks like 'http://foo:bar@127.0.0.1:5984'

Thanks again to both of you :)
